### PR TITLE
[Please Test] rewrite of the event class and the database

### PR DESCRIPTION
**Pull request by geier**
this should fix some issues (like #222 and #195) and enable the deletion and modification of single instances of recurring events (not implemented yet). 

and yes, I know some tests are failling.

 * [x] enable simple sorting of events
 * [x] print warning when using default_timezone
 * [ ] support for RDATE, EXDATE, RRULE:UNTIL with invalid timezone information
 * [x] re-fix RRULE editing

### CODE REVIEW

File: khal/khalendar/event.py [hunk 1]
```diff
@@ -132,143 +208,124 @@ def symbol_strings(self):
...
+        """neeeded for vdirsyncer compat"""
+        return self._vevents[self.ref]['UID']
 
-        return '{}: {}{}{}{}'.format(
-            rangestr, self.summary, location, repitition, description)
+    @property
+    def summary(self):
+        return self._vevents[self.ref].get('SUMMARY', '')
+
+    def update_summary(self, summary):
```
Comment (untitaker) [inline khal/khalendar/event.py]: Use

    @summary.setter
    def summary(self, summary):
        ...

File: khal/khalendar/event.py [hunk 2]
```diff
@@ -132,143 +208,124 @@ def symbol_strings(self):
...
+        self._vevents[self.ref]['SUMMARY'] = summary
+
+    @property
+    def location(self):
+        return self._vevents[self.ref].get('LOCATION', '')
+
+    def update_location(self, location):
```
Comment (untitaker) [inline khal/khalendar/event.py]: Use setter.

File: khal/khalendar/event.py [hunk 3]
```diff
@@ -132,143 +208,124 @@ def symbol_strings(self):
...
+        self._vevents[self.ref]['LOCATION'] = location
+
+    @property
+    def description(self):
+        return self._vevents[self.ref].get('DESCRIPTION', '')
 
-    def compact(self, day):
+    def update_description(self, description):
```
Comment (untitaker) [inline khal/khalendar/event.py]: Use setter.

File: khal/khalendar/event.py (hunk 1 above)
Comment (geier) [inline khal/khalendar/event.py]: while this might make sense at the moment, all those update methods will soon get a `ref` and `thisandfuture` flag.

Comment (untitaker) [inline khal/khalendar/event.py]: How would you get the summary of *past* events? Is there an `Event` instance for each instance, and do these share all the same `self._vevents`?

Comment (untitaker) [inline khal/khalendar/event.py]: Perhaps what we need is an additional class called `Recurrence`, so `self._vevents` is a list of `Recurrence`-instances. This class then has all the properties (`location`, `summary`).

Comment (geier) [inline khal/khalendar/event.py]: Yes, an Event represents only one specific recurrence instance and all recurrence instances of the same proto Event have the same self._vevents (but not the identical object). I don't believe that with the current functionality of khal we really need to access other recurrence instances properties (if they are deviating).

**Comment by geier** [comment]
while this isn't as finished yet as I would like, I'm merging now because it's better than before and I hopefully have the time to fix all new bugs in the next week.

**Pull request merged**